Core/include/Acts/Utilities/BinAdjustment.hpp

```diff
@@ -121,6 +124,54 @@ BinUtility adjustBinUtility(const BinUtility& bu, const CylinderBounds& cBounds,
   }
   return uBinUtil;
 }
+/// @brief adjust the BinUtility bu to the dimensions of planar bounds
+///
+/// @param bu BinUtility at source
+/// @param rBounds the Rectangular bounds to adjust to
+///
+/// @return new updated BinUtiltiy
+BinUtility adjustBinUtility(const BinUtility& bu,
+                            const RectangleBounds& rBounds,
+                            const Transform3& transform) {
+  // Default constructor
+  BinUtility uBinUtil(transform);
+
+  // The parameters from the cylinder bounds
+  double minX = rBounds.get(RectangleBounds::eMinX);
+  double minY = rBounds.get(RectangleBounds::eMinY);
+  double maxX = rBounds.get(RectangleBounds::eMaxX);
+  double maxY = rBounds.get(RectangleBounds::eMaxY);
+
+  // Retrieve the binning data
+  const std::vector<BinningData>& bData = bu.binningData();
+  // Loop over the binning data and adjust the dimensions
+  for (auto& bd : bData) {
+    // The binning value
+    BinningValue bval = bd.binvalue;
+    // Throw exceptions is stuff doesn't make sense:
+    // - not the right binning value
+    // - not equidistant
+    if (bd.type == arbitrary) {
+      throw std::invalid_argument("Arbirary binning can not be adjusted.");
+    }  // else if (bval != binR and bval != binPhi) {
+       // std::cout << "bval = " << bval << std::endl;
+       // throw std::invalid_argument("Disc binning must be: phi, r");
+       //  }
+    float min, max = 0.;
+    // Perform the value adjustment
+    if (bval == binPhi) {
```

Comment:
I do not understand why this is `binPhi` - shouldn't the adjustment be in x and y ?